src/pykeen/nn/node_piece.py

```diff
@@ -0,0 +1,683 @@
+    )
+
+
+class AnchorSelection:
+    """Anchor entity selection strategy."""
+
+    def __init__(self, num_anchors: int = 32) -> None:
+        """
+        Initialize the strategy.
+
+        :param num_anchors:
+            the number of anchor nodes to select.
+            # TODO: allow relative
+        """
+        self.num_anchors = num_anchors
+
+    @abstractmethod
+    def __call__(self, edge_index: numpy.ndarray) -> numpy.ndarray:
+        """
+        Select anchor nodes.
+
+        .. note ::
+            the number of selected anchors may be smaller than $k$, if there
+            are less entities present in the edge index.
+
+        :param edge_index: shape: (m, 2)
+            the edge_index, i.e., adjacency list.
+
+        :return: (k,)
+            the selected entity ids
+        """
+        raise NotImplementedError
+
+    def extra_repr(self) -> Iterable[str]:
+        """Extra components for __repr__."""
+        yield f"num_anchors={self.num_anchors}"
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}({', '.join(self.extra_repr())})"
+
+
+class DegreeAnchorSelection(AnchorSelection):
+    """Select entities according to their (undirected) degree."""
+
+    def __call__(self, edge_index: numpy.ndarray) -> numpy.ndarray:  # noqa: D102
+        unique, counts = numpy.unique(edge_index, return_counts=True)
+        top_ids = numpy.argpartition(counts, max(counts.size - self.num_anchors, 0))[-self.num_anchors :]
+        return unique[top_ids]
+
+
+class PageRankAnchorSelection(AnchorSelection):
+    """Select entities according to their page rank."""
+
+    def __init__(
+        self,
+        num_anchors: int = 32,
+        max_iter: int = 1_000,
+        alpha: float = 0.05,
+        epsilon: float = 1.0e-04,
+    ) -> None:
+        """
+        Initialize the selection strategy.
+
+        :param num_anchors:
+            the number of anchors to select
+        :param max_iter:
+            the maximum number of power iterations
+        :param alpha:
+            the smoothing value / teleport probability
+        :param epsilon:
+            a constant to check for convergence
+        """
+        super().__init__(num_anchors=num_anchors)
+        self.max_iter = max_iter
+        self.alpha = alpha
+        self.beta = 1.0 - alpha
+        self.epsilon = epsilon
+
+    def extra_repr(self) -> Iterable[str]:  # noqa: D102
+        yield from super().extra_repr()
+        yield f"max_iter={self.max_iter}"
+        yield f"alpha={self.alpha}"
+        yield f"epsilon={self.epsilon}"
+
+    def __call__(self, edge_index: numpy.ndarray) -> numpy.ndarray:  # noqa: D102
+        # convert to sparse matrix
+        adj = _edge_index_to_sparse_matrix(edge_index=edge_index)
+        # symmetrize
+        # TODO: should we add self-links
+        # adj = (adj + adj.transpose() + scipy.sparse.eye(m=adj.shape[0], format="coo")).tocsr()
+        adj = (adj + adj.transpose()).tocsr()
```

Comment:
IIRC, we don't need self-loops for PageRank.

We could even drop the inverse edges, but for numerical stability (to prevent sinks) of the power iteration I think we can keep them